src/sage/topology/moment_angle_complex.py

```diff
@@ -0,0 +1,781 @@
+            True
+        """
+        return self._components
+
+    def dimension(self):
+        r"""
+        The dimension of ``self``.
+
+        The dimension of a moment-angle complex is the dimension
+        of the constructed (cubical) complex. It is not difficult to
+        see that this turns out to be `m+n+1`, where `m` is the number
+        of vertices and `n` is the dimension of the associated simplicial
+        complex.
+
+        EXAMPLES::
+
+            sage: Z = MomentAngleComplex([[0,1], [1,2,3]])
+            sage: Z.dimension()
+            7
+            sage: Z = MomentAngleComplex([[0, 1, 2]])
+            sage: Z.dimension()
+            6
+            sage: dim(Z)  # indirect doctest
+            6
+
+        We can construct the cubical complex and compare whether
+        the dimensions coincide::
+
+            sage: dim(Z) == dim(Z.cubical_complex())
+            True
+        """
+        number_of_vertices = len(self._simplicial_complex.vertices())
+        dim = self._simplicial_complex.dimension()
+        return number_of_vertices + dim + 1
+
+    @cached_method  # maybe ignore the algorithm?
+    def _homology_group(self, i, base_ring, cohomology, algorithm, verbose, reduced):
+        """
+        The `i`-th (reduced) homology group of ``self``.
+
+        .. SEEALSO::
+
+            :meth:`homology`,
+            :meth:`.cell_complex.GenericCellComplex.homology`.
+
+        TESTS::
+
+            sage: Z = MomentAngleComplex([[0,1,2], [1,2,3]]); Z
+            Moment-angle complex of Simplicial complex with vertex set (0, 1, 2, 3) and facets {(0, 1, 2), (1, 2, 3)}
+            sage: Z._homology_group(3, base_ring=ZZ, cohomology=False, algorithm='pari', verbose=False, reduced=True)
+            Z
+            sage: Z._homology_group(4, base_ring=ZZ, cohomology=False, algorithm='pari', verbose=False, reduced=True)
+            0
+            sage: Z.homology()
+            {0: 0, 1: 0, 2: 0, 3: Z, 4: 0, 5: 0, 6: 0, 7: 0}
+            sage: RP = simplicial_complexes.RealProjectivePlane()
+            sage: Z = MomentAngleComplex(RP)
+            sage: Z._homology_group(8, base_ring=ZZ, cohomology=False, algorithm='pari', verbose=False, reduced=True)
+            C2
+
+        This yields the same result as creating a cubical complex
+        from this moment-angle complex, and then computing its (co)homology,
+        but that is incomparably slower and is really only possible when
+        the associated simplicial complex is very small::
+
+            sage: Z = MomentAngleComplex([[0,1], [1,2], [2,0]]); Z
+            Moment-angle complex of Simplicial complex with vertex set (0, 1, 2) and facets {(0, 1), (0, 2), (1, 2)}
+            sage: Z.cubical_complex()
+            Cubical complex with 64 vertices and 729 cubes
+            sage: Z.cubical_complex().homology(5) == Z._homology_group(5, base_ring=ZZ, cohomology=False, algorithm='pari', verbose=False, reduced=True)
+            True
```

Comment:
It's not worthwhile to reproduce tests/docstrings. I would test stuff specific to this method.